classes/mail/variables/QueryEmailVariable.inc.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @file classes/mail/variables/QueryEmailVariable.inc.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class QueryEmailVariable
+ * @ingroup mail_variables
+ *
+ * @brief Represents variables associated with a query
+ */
+
+namespace PKP\mail\variables;
+
+use PKP\query\Query;
+
+class QueryEmailVariable extends Variable
+{
+    public const SUBJECT = 'querySubject';
+    public const LAST_NOTE_CONTENT = 'lastContent';
+
+    protected Query $query;
+
+    public function __construct(Query $query)
+    {
+        $this->query = $query;
+    }
+
+    /**
+     * @copydoc Variable::descriptions()
+     */
+    public static function descriptions(): array
+    {
+        return
+        [
+            self::SUBJECT => __('emailTemplate.variable.query.subject'),
+            self::LAST_NOTE_CONTENT => __('emailTemplate.variable.query.lastContent'),
+        ];
+    }
+
+    /**
+     * @copydoc Variable::values()
+     */
+    public function values(string $locale): array
+    {
+        $notes = $this->query->getReplies(null, \PKP\note\NoteDAO::NOTE_ORDER_DATE_CREATED, \PKP\db\DAO::SORT_DIRECTION_DESC, true);
+        $lastNote = $notes->next();
+
+        return
+        [
+            self::SUBJECT => $this->query->getHeadNote()->getTitle(),
+            self::LAST_NOTE_CONTENT => $lastNote->getContents(),
+        ];
+    }
```

Comment:
I'd prefer to add notes as something like read-only property to the `Query` object and set it before the latter is passed to the Mailable's constructor.

Reply:
Or keep it simple, use `Query::setData()`

Reply:
So you mean something like this?

```php
$query->setData();
$mailable = new ExampleMailable($query);
```

Reply:
Yes

Reply:
I'm worried that it will create confusion with the type hinting. The `ExampleMailable` can type hint that it wants a `Query` object, but it has no way of saying that it expects `$query->setData()` to be run before it is passed in.

I'm leaning towards something like this in the mailable:

```php
public const SUBJECT = 'subject';
public const MESSAGE = 'message';

/**
 * @param array $notes All notes in the query, ordered by most recent first
 * @param string $subject The query's subject
 */
public function __construct(Context $context, PKPSubmission $submission, array $notes, string $subject)
{
    parent::__construct(array_slice(func_get_args(), 0, -2));

    $lastNote = end($notes);
    
    $this->addData([
        self::SUBJECT => $this->subject,
        self::MESSAGE => $lastNote->getContents(),
    ]);
}

public static function getDataDescriptions(): array
{
    $variables = parent::getDataDescriptions();
    $variables[self::SUBJECT] = __('...');
    $variables[self::MESSAGE] = __('...');
    return $variables;
}
```

What do you think?

Reply:
Just for information, you can't use variable named 'message' in template because Laravel populate this variable
https://laravel.com/docs/9.x/mail#inline-attachments
So I named it 'content'